### PR TITLE
RasterBand::histogram(): error out if n_buckets == 0 to avoid crash in GDAL

### PR DESCRIPTION
**Pull request by rouault**
…
- [X] I agree to follow the project's [code of conduct](https://github.com/georust/gdal/blob/master/CODE_OF_CONDUCT.md).
- [ ] I added an entry to `CHANGES.md` if knowledge of this change could be valuable to users.
---

### CODE REVIEW

File: src/raster/tests.rs [hunk 1]
```diff
@@ -782,6 +782,12 @@ fn test_raster_histogram() {
         .map(|i| expected[i] + expected[i + 1])
         .collect::<Vec<_>>();
     assert_eq!(hist.counts(), &expected_small);
+
+    // n_buckets = 0 is not allowed
+    match rb.histogram(-0.5, 255.5, 0, true, true) {
+        Err(_) => (), // error expected
+        Ok(_) => panic!("Expected Err, got Ok")
```
Comment (lnicola) [inline src/raster/tests.rs]: You can use unwrap_err to make this simpler, or match against the expected error variant to make the test more robust.

Comment (lnicola) [reply src/raster/tests.rs]: (sorry, that's `expect_err`)

Comment (rouault) [reply src/raster/tests.rs]: thanks, commit updated

File: src/raster/tests.rs [hunk 2]
```diff
@@ -782,6 +782,10 @@ fn test_raster_histogram() {
         .map(|i| expected[i] + expected[i + 1])
         .collect::<Vec<_>>();
     assert_eq!(hist.counts(), &expected_small);
+
+    // n_buckets = 0 is not allowed
+    let hist = rb.histogram(-0.5, 255.5, 0, true, true);
+    hist.expect_err("n_buckets should be > 0");
```
Comment (lnicola) [inline src/raster/tests.rs]:
```suggestion
    hist.expect_err("histogram with 0 buckets should panic");
```

Comment (rouault) [reply src/raster/tests.rs]: thanks. updated

**Pull request closed without merging**